compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2178,6 +2175,9 @@ public void visit(BLangSimpleVarRef varRefExpr) {
                     actualType = symTable.semanticError;
                     dlog.error(varRefExpr.pos, DiagnosticErrorCode.CANNOT_UPDATE_CONSTANT_VALUE);
                 }
+            } else if ((symbol.tag & SymTag.TYPE_DEF) == SymTag.TYPE_DEF || (symbol.tag & SymTag.TYPE) == SymTag.TYPE) {
```

Comment:
I'm not sure if adding a check for SymTag.TYPE here is correct. IINM, SymTag.TYPE_DEF should have been set for the problem cases here.

@pubudu91 thoughts please?

Reply:
What prompted to make this change? I also don't feel like checking for the `TYPE` sym tag here is correct.

Reply:
```ballerina
function testEnsureTypeFunction() {
    int number = 10;
    int:Unsigned32|error number1 = number.ensureType(int:Unsigned32);
}
```
The expected type for the parameter of `ensureType` is `typedesc`. Here `int:Unsigned32` is a simple variable reference. I used above change to set `actualType` as `typedesc<int:Unsigned32>`

Reply:
The original issue was that we couldn't refer to a builtin-subtype in a context where a typedesc is expected.

e.g.,
```ballerina
typedesc td = int:Unsigned32; // undefined symbol 'Unsigned32'
```

I just checked this with a user-defined type in a different module and it works without an issue. I think this can be an issue with the symbols defined specifically for builtin-subtypes (we create the symbol manually even though the type is defined in the langlib). Ideally, the symbol should have had the TYPE_DEF tag. 

@chiranSachintha, can you please check?

Reply:
This isn't an issue only via the BIR btw. Trying to refer to the type within the relevant langlib also fails with the same error.

Reply:
`builtin-subtypes` only contains the TYPE tag. Should it have a `TYPE_DEF` tag though it is not a type definition?

Reply:
Right now, I believe for types associated with a type definition, we also use the TYPE_DEF tag. You can check this with something like

```ballerina
type Foo record {|
    
|};

Foo f = {}; // check the tag for Foo's type symbol here
```

I think we'll have to do the same here at least for now.